factory/processFactory.go

```diff
@@ -79,3 +81,33 @@ func getPubKeyConverter(cfg config.GeneralConfig) (core.PubkeyConverter, error)
 		return nil, common.ErrInvalidPubKeyConverterType
 	}
 }
+
+// CreatePayloadHandler will create a new instance of payload handler
+func CreatePayloadHandler(connType string, marshaller marshal.Marshalizer, wsMarshaller marshal.Marshalizer, facade process.EventsFacadeHandler) (websocket.PayloadHandler, error) {
```

Comment:
Instead of having `marshaller marshal.Marshalizer, wsMarshaller marshal.Marshalizer` (in which case you need to fill both either if you want ws/http), we could have a `marshallerType string` or simply just a `marshaller marshal.Marshalizer`

Reply:
refactored to pass only configs, fix for a separate marshaller for header handling